napari/layers/surface/surface.py

```diff
@@ -282,14 +289,37 @@ def gamma(self, value):
     def _set_view_slice(self):
         """Sets the view given the indices to slice with."""
 
-        not_disp = list(self.dims.not_displayed)
-        disp = list(self.dims.displayed)
-        indices = np.array(self.dims.indices)
+        # Take vertex_values dimensionality into account if more than one value
+        # is provided per vertex.
+        if self.vertex_values.ndim > 1:
+            values = self.vertex_values[
+                self.dims.indices[: -self.vertices.shape[1]]
+            ]
```

Comment:
Ok so it took me quite a while to understand this, and I think it's key to the whole process. If I understand correctly, the dims model has L + 3 dimensions. (or L + D, rather). So now we want to grab the current values, so we get the index for axes [0, 1, ..., -D]. I suggest a comment to this effect!

Reply:
added